code/modules/station_goals/meteor_shield.dm

```diff
@@ -94,34 +95,42 @@
 			return FALSE
 	return TRUE
 
-/obj/machinery/satellite/meteor_shield/process()
-	if(obj_flags & EMAGGED)
-		//kills the processing because emagged meteor shields no longer stop meteors in any way
-		return PROCESS_KILL
-	if(!active)
-		return
-	for(var/obj/effect/meteor/meteor_to_destroy in GLOB.meteor_list)
-		if(meteor_to_destroy.z != z)
-			continue
-		if(get_dist(meteor_to_destroy, src) > kill_range)
-			continue
+/obj/machinery/satellite/meteor_shield/Initialize(mapload)
+	. = ..()
+	proximity_monitor = new(src,0)
+
+/obj/machinery/satellite/meteor_shield/HasProximity(atom/movable/proximity_check_mob)
+	. = ..()
+	if(istype(proximity_check_mob, /obj/effect/meteor))
+		var/obj/effect/meteor/meteor_to_destroy = proximity_check_mob
 		if(space_los(meteor_to_destroy))
 			var/turf/beam_from = get_turf(src)
 			beam_from.Beam(get_turf(meteor_to_destroy), icon_state="sat_beam", time = 5)
 			if(meteor_to_destroy.shield_defense(src))
 				qdel(meteor_to_destroy)
 
 /obj/machinery/satellite/meteor_shield/toggle(user)
+	if(user)
+		src.balloon_alert(user, "you're looking for the [active ? "off" : "on"] button on the [src]")
+	if(user && !do_after(user, 2 SECONDS, src, IGNORE_HELD_ITEM))
+		return FALSE;
```

Comment:
```suggestion
		return FALSE
```
DM oft doesn't need a statement separator on new lines outside of the precompiler iirc.